services/portal/GWASUIApp/GWASUIAppTasks.js

```
@@ -130,4 +135,91 @@ module.exports = {
+        const wfName = data.name;
+        if (process.env.DEBUG === 'true') {
+            console.log(`### workflow name : ${wfName}`);
+        };
+        return wfName;
+    },
+
+    async getWFUID() {
+        const data = await this.getUserWF();
+        const wfUID = data.uid;
+        if (process.env.DEBUG === 'true') {
+            console.log(`### workflow uid : ${wfUID}`);
+        };
+        return wfUID;
+    },
+
+   async getWFStatus() {
+        const data = await this.getUserWF();
+        const wfStatus = data.phase;
+        if (process.env.DEBUG === 'true') {
+            console.log(`### workflow status : ${wfStatus}`);
+        };
+        return wfStatus;
+    },
+
+    executionButton() {
+        I.saveScreenshot('executionButton.png');
+        I.click(GWASUIAppProps.ExecutionButton);
+        I.wait(5);
+        I.seeElement(GWASUIAppProps.executionView);
+        I.seeElement(GWASUIAppProps.executionLogs);
+        I.click(GWASUIAppProps.backButton);
+    },
+
+    resultsButton() {
+        I.saveScreenshot('resultButton.png');
+        I.click(GWASUIAppProps.ResultsButton);
+        I.wait(5);
+        I.seeElement(GWASUIAppProps.ResultsButton);
+        I.seeElement(GWASUIAppProps.manhattanPlot);
+        I.seeElement(GWASUIAppProps.downloadAllResults);
+        I.click(GWASUIAppProps.resultsBackButton);
+    },
+
+    actionButton() {
+        I.seeElement(GWASUIAppProps.actionButton);
+        I.click(GWASUIAppProps.actionButton);
+        I.seeElement(GWASUIAppPro.downloadAllResults);
+    },
+
+    async completeWFCheck() {
+        const workflowName = await this.getWFName();
+        const status = await this.getWFStatus();
+        if (status === 'Succeeded') {
+            console.log(`### Status : ${status}.Workflow ${workflowName} has completed successfully`);
+            this.executionButton();
+            this.resultsButton();
+        } else if (status === 'Error' || status === 'Failed') {
```

should we let the test fail in this case? I think so.